crates/core-executor/src/query.rs

```
@@ -381,8 +383,14 @@ impl UserQuery {
                 Statement::CopyIntoSnowflake { .. } => {
                     return Box::pin(self.copy_into_snowflake_query(*s)).await;
                 }
-                Statement::AlterTable { .. }
-                | Statement::StartTransaction { .. }
+                Statement::AlterTable { .. } => {
+                    NotSupportedStatementSnafu {
+                        statement: "ALTER TABLE".to_string(),
+                    }
+                    .fail()?;
+                    return self.status_response();
```

Why return the `self.status_responce`, if the error is guaranteed to bubble up from there? You could probably return the error if the compiler is asking for a return there.